velox/vector/fuzzer/VectorFuzzer.cpp

```diff
@@ -715,12 +763,22 @@ TypePtr VectorFuzzer::randType(int maxDepth) {
       VARBINARY(),
       TIMESTAMP(),
       DATE(),
-  };
-  static constexpr int kNumScalarTypes =
-      sizeof(kScalarTypes) / sizeof(kScalarTypes[0]);
+      // Precision and scale are just placeholders and will be
+      // replaced with random values.
+      SHORT_DECIMAL(ShortDecimalType::kMaxPrecision, 0),
+      LONG_DECIMAL(LongDecimalType::kMaxPrecision, 0)};
+  auto scalarTypes =
+      supportedScalarTypes.size() == 0 ? kScalarTypes : supportedScalarTypes;
+  const int numScalarTypes = sizeof(scalarTypes) / sizeof(scalarTypes[0]);
   // Should we generate a scalar type?
   if (maxDepth <= 1 || rand<bool>(rng_)) {
-    return kScalarTypes[rand<uint32_t>(rng_) % kNumScalarTypes];
+    auto type = scalarTypes[rand<uint32_t>(rng_) % numScalarTypes];
```

Comment:
This logic is confusing to me. I wonder if a cleaner design would be to have a list of scalar type kinds and have a method to generate a type for a given type kind. The supportedScalarTypes parameter would also be a list of type kinds, not types.

Reply:
That sounds better what we have now. I will add that change.